Refactor SCIM provisioning documentation

irenerl24 opened the pull request:
Removed redundant admonitions and clarified SCIM provisioning warning.

<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/main/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the main branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

7. If your PR content should be added to the What's New document for the next major or minor release, add the **add to what's new** label to your PR. Note that you should add this label to the main PR that introduces the feature; do not add this label to smaller PRs for the feature.

-->

**What is this feature?**

[Add a brief description of what the feature or update does.]

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

irenerl24 committed:
Refactor SCIM provisioning documentation

Removed redundant admonitions and clarified SCIM provisioning warning.

linoman commented on docs/sources/setup-grafana/configure-security/configure-scim-provisioning/_index.md:
```diff
@@ -22,19 +22,17 @@ System for Cross-domain Identity Management (SCIM) is an open standard that allo
 Grafana Labs offers limited support, and breaking changes might occur prior to the feature being made generally available.
-{{< /admonition >}}
 
-{{< admonition type="warning" >}}
 **Public Preview:** SCIM provisioning is currently in Public Preview. While functional, the feature is actively being refined and may undergo changes. We recommend thorough testing in non-production environments before deploying to production systems.
-{{< /admonition >}}
```
`suggestion` Let's keep this warning until we're ready to migrate to GA.

irenerl24 committed:
Restore critical SAML and SCIM alignment warnings

Reinstate critical information about aligning SAML identifiers with SCIM externalId and emphasize the importance of stable identifiers for security.

linoman commented:
While this is approved, I will need to revisit the docs. The definition of the ExternalID has changed, and not is opt-in instead of opt-out.
